Improve the concept of "resources" in the RDMkit

bedroesb opened the pull request:
This will close #1518 
- [x] Choosing a more neutral icon for the tools in the text
   ![image](https://github.com/user-attachments/assets/d7888812-5538-40f0-9abe-e060be80202b)
- [x] Define what resources are with examples in our contribute documentation

bianchini88 commented:
@bedroesb I do not think it is up to me to approve the new icon. If we really want to close https://github.com/elixir-europe/rdmkit/issues/1518 we should also expand the documentation. I can have a go at it, but not before the 17th of October.

bedroesb commented:
I agree, this needs more thought than just the icon.

bedroesb commented:
Based on the editorial meetings, an info icon fits best the purpose of the clickable 'buttons'. 
Two new variants:
![image](https://github.com/user-attachments/assets/6b60adfa-4c21-484a-ba3d-9b9bd1125589)
![image](https://github.com/user-attachments/assets/3a3dd848-cab3-4fd4-b403-490e1fbeaf45)

bianchini88 commented:
I do like the ones with the colour filling and intermediate "i" sizes.

bedroesb commented:
@bianchini88 I've added some numbers, makes it easier :) I am also adding a paragraph to the 'Update a tool or resource''

bianchini88 commented:
Thanks @bedroesb, I actually struggled to understand how many variations we had. I would say 3>1>2>4 (but please ask someone with a better graphic sense than me).

bedroesb commented:
I added 2 more, my current favorites are 3 and 6

bedroesb commented:
@elixir-europe/rdmkit-editors I added 6 options for an icon above and updated the **Add a new tool or resource** contribute page be more clear about when a tool should be added to the tools table. Did I forget things or remember things wrongly? Please do give feedback. I might also use the word `main YAML file` to much ....

Nazeeefa commented:
I am afraid I can't vote for any of the icon options for tools and resources because I like the current gear icon 😄 The "i" icon suggests, to me at least, that hovering over icon will give me a brief information and I do not expect anything more.

diana-pilvar commented:
if only used for tools, I like the gear. If the links also cover resources, gear is not optimal. 
I don't like the box.
For the I's : 4,3 >1>2,4
6 reminds me of biohazard

bianchini88 commented on pages/contribute/tool_resource_update.md:
```diff
@@ -35,15 +50,11 @@ Example:
-## Making changes
-
-1. Make sure the tool you want to add is not already described in the [yaml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml). If so, go to step 3, if not, follow the next step.
+1. Make sure the tool you want to add is not already described in the [YAML file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml). If so, go to step 3, if not, follow the next step.
```
```suggestion
1. Make sure the tool you want to add is not already listed in the [YAML file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml). If so, go to step 3, if not, follow the next step.
```

korbinib commented:
1>3>2>4

bedroesb commented:
In the editorial meeting we voted on icon number 3. This is put in place now. @bianchini88  I think I incorporated all your feedback, what do you think?

bedroesb commented:
@elixir-europe/rdmkit-editors The new logo is in place and our documentation about what is a resource is improved. This PR is ready I think!
